stdlib/ballerina-config/src/main/java/org/ballerinalang/config/cipher/AESCipherTool.java

```diff
@@ -0,0 +1,137 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.config.cipher;
+
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * This tools is used to encrypt and decrypt data using AES Algorithm CBC mode and PKCS #5 padding.
+ *
+ * @since 0.964.0
+ */
+public class AESCipherTool {
+
+    private static final String ALGORITHM_AES_CBC_PKCS5 = "AES/CBC/PKCS5Padding";
+    private static final String ALGORITHM_AES = "AES";
+    private static final String ALGORITHM_SHA_256 = "SHA-256";
+    private static final String DEFAULT_INITIALIZATION_VECTOR = "0123456789ABCDEF";
+
+    private final Cipher encryptionCipher;
+    private final Cipher decryptionCipher;
+
+    /**
+     * @param userSecret User secret String to encode and decode a value.
+     * @throws AESCipherToolException if Any error occurs when preparing the tool with given user data.
+     */
+    public AESCipherTool(String userSecret) throws AESCipherToolException {
+        this(userSecret, null);
```

Comment:
We should not use the hard-coded IV by default. Have a look at [1] on recommendations. We should ideally use a fixed size random IV each time and then append the random IV with the encrypted data. Now, during decryption, since we know the size of IV, we can extract is separately. Current code will produce same cipher-text for same-plain-text, over and over again. Since, we do not have enough randomness, it will be easier to derive the encryption-key by processing bunch of encrypted strings. 

[1] https://stackoverflow.com/questions/31238352/java-aes-encryption-and-decryption-with-static-secret

Reply:
Understood. Will make the changes. :)